editor/assets/default_renderpipeline/builtin-dof-pass.ts

```diff
@@ -0,0 +1,323 @@
+
+        if (cameraConfigs.enableDof) {
+            // Output scene depth, this is allowed but has performance impact
+            cameraConfigs.enableStoreSceneDepth = true;
+            ++cameraConfigs.remainingPasses;
+        }
+    }
+    windowResize(
+        ppl: rendering.BasicPipeline,
+        pplConfigs: Readonly<PipelineConfigs>,
+        cameraConfigs: Readonly<CameraConfigs & DofPassConfigs>,
+        window: renderer.RenderWindow,
+        camera: renderer.scene.Camera,
+        nativeWidth: number,
+        nativeHeight: number): void {
+        const id = window.renderWindowId;
+        if (cameraConfigs.enableDof) {
+            ppl.addRenderTarget(`DofRadiance${id}`,
+                cameraConfigs.radianceFormat,
+                cameraConfigs.width,
+                cameraConfigs.height);
+        }
+    }
+    setup(
+        ppl: rendering.BasicPipeline,
+        pplConfigs: Readonly<PipelineConfigs>,
+        cameraConfigs: CameraConfigs & Readonly<DofPassConfigs>,
+        camera: renderer.scene.Camera,
+        context: PipelineContext,
+        prevRenderPass?: rendering.BasicRenderPassBuilder): rendering.BasicRenderPassBuilder | undefined {
+        if (!cameraConfigs.enableDof) {
+            return prevRenderPass;
+        }
+        --cameraConfigs.remainingPasses;
+
+        if (cameraConfigs.remainingPasses === 0) {
+            return this._addDepthOfFieldPasses(ppl, pplConfigs,
+                cameraConfigs, this._material,
+                camera, cameraConfigs.width, cameraConfigs.height,
+                context.colorName,
+                context.depthStencilName,
+                cameraConfigs.colorName);
+        } else {
+            const prefix = cameraConfigs.enableShadingScale
+                ? `ScaledRadiance`
+                : `Radiance`;
+            const outputRadianceName = getPingPongRenderTarget(
+                context.colorName, prefix, cameraConfigs.renderWindowId);
+            const inputRadianceName = context.colorName;
+            context.colorName = outputRadianceName;
+            return this._addDepthOfFieldPasses(ppl, pplConfigs,
+                cameraConfigs, this._material,
+                camera, cameraConfigs.width, cameraConfigs.height,
+                inputRadianceName,
+                context.depthStencilName,
+                outputRadianceName);
+        }
+    }
+    private _addDepthOfFieldPasses(
+        ppl: rendering.BasicPipeline,
+        pplConfigs: Readonly<PipelineConfigs>,
+        cameraConfigs: CameraConfigs & Readonly<DofPassConfigs>,
+        dofMaterial: Material,
+        camera: renderer.scene.Camera,
+        width: number,
+        height: number,
+        inputRadiance: string,
+        inputDepthStencil: string,
+        outputRadianceName: string,
+    ): rendering.BasicRenderPassBuilder {
+        this._cocParams.x = this._minRange;
+        this._cocParams.y = this._maxRange;// camera.farClip;// this._focusRange;
```

Comment:
What does the comment `// camera.farClip;// this._focusRange;` mean?
Could it be removed?

Reply:
It can be removed.